Sources/MapboxCoreNavigation/NavigationServiceDelegate.swift

```diff
@@ -98,6 +98,8 @@ public protocol NavigationServiceDelegate: AnyObject, UnimplementedLogging {
      
      You can implement this method to allow the navigation service to continue check and reroute the user if needed. By default, the user will not be rerouted when arriving at a waypoint.
      
+     - important: Only works for `LegacyRouteController`.
```

Comment:
This delegate method is likely in use in the wild, so the release notes should mention that it no longer works with RouteController and offer an alternative.

LegacyRouteController is marked as deprecated, so we could communicate this new behavior by marking the delegate method as deprecated too. However, the compiler doesn’t warn when you implement a deprecated delegate method, only when we call it.

Reply:
Updated the changelog and marked related entities deprecated